ament_index_cpp/src/has_resource.cpp

```diff
@@ -35,19 +35,34 @@ has_resource(
   if (resource_name.empty()) {
     throw std::runtime_error("ament_index_cpp::has_resource() resource name must not be empty");
   }
+  std::optional<std::filesystem::path> result = is_resource_available(resource_type, resource_name);
+  if (result.has_value()) {
+    if (prefix_path) {
+      *prefix_path = result.value().c_str();
+    }
+    return true;
+  }
+  return false;
+}
+
+std::optional<std::filesystem::path>
+is_resource_available(
+  const std::string & resource_type,
+  const std::string & resource_name)
+{
+  std::optional<std::filesystem::path> result = std::nullopt;
+
   auto paths = get_search_paths();
   for (auto path : paths) {
     auto resource_path = path + "/share/ament_index/resource_index/" +
       resource_type + "/" + resource_name;
```

Comment:
```suggestion
    auto resource_path = path / "share" / "ament_index" / "resource_index" /
      resource_type / resource_name;
```
Would it be time to also do changes like this?